src/test_utils.rs

```diff
@@ -340,7 +343,18 @@ impl TestContextBuilder {
                 .addr;
             let name = EmailAddress::new(&addr).unwrap().local;
 
-            let test_context = TestContext::new_internal(Some(name), self.log_sink).await;
+            let mut unused_name = name.clone();
+            if let Some(used_names) = used_names {
+                // If there are multiple Alices, call them 'alice', 'alice2', 'alice3', ...
+                let mut i = 1;
+                while used_names.contains(&unused_name) {
```

Comment:
Maybe make it a map instead of looping, i.e. insert ("alice", 1) when this called the first time, etc.? We don't have numbers in names by default, so it will work